### PR TITLE
add remove table feature

### PR DESCRIPTION
**Pull request by alexey-petrov**
Closes:
- https://github.com/sparkletown/internal-sparkle-issues/issues/1633

* Added "remove table" button to `jazzbar`@`music venue` and `conversationSpace`
* Reused modal for delete confirmation
* Minor table UI fixes
* Avaialble to admin users only

image
image
image

### CODE REVIEW

**Review by goran-peoski-work — APPROVED** [review]
Plenty more comments I can add, somehow I doubt they can all fit in the scope of this PR.

Approving with an advice to go over the comments and see if few of them about the logic, data and FE-BE communication, should be acted upon.

File: src/utils/venue.ts
Comment (alexey-petrov) [reply src/utils/venue.ts]: done ea78f9d

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: done bd10ed1

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: I've moved table update logic into `removeTable` cloud function and got rid of `JazzBarTableComponent` entirely, so that now there's only one table component bd10ed1

File: src/components/molecules/TableComponent/TableComponent.scss
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.scss]: done bd10ed1

File: src/components/molecules/TableComponent/TableComponent.scss
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.scss]: done bd10ed1

File: functions/venue.js
Comment (alexey-petrov) [reply functions/venue.js:806]: done bd10ed1

File: functions/venue.js
Comment (alexey-petrov) [reply functions/venue.js]: done bd10ed1

File: src/api/table.ts
Comment (alexey-petrov) [reply src/api/table.ts]: done bd10ed1

File: src/api/table.ts
Comment (alexey-petrov) [reply src/api/table.ts]: done bd10ed1

File: functions/venue.js
Comment (alexey-petrov) [reply functions/venue.js]: changes some of the names in bd10ed1

File: functions/venue.js
Comment (alexey-petrov) [reply functions/venue.js]: done bd10ed1

File: src/components/molecules/TableComponent/TableComponent.scss
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.scss]: fixed styles in this file de57b81

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:69]: No gain of a simple alias, might as well just use `isDeletingTable` for the button prop
```suggestion
```

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:86]: Static values. This can safely be moved to SCSS file and just flip two classes like `TableComponent--jazzbar`, `TableComponent--not-jazzbar`

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:94]: Is it possible to prefix this?

```suggestion
    <div className={`TableComponent TableComponent--ref-${table.reference}`}>
```

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:97]: Leave it to Prettier to make code uglier 🤦

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:120]: There is a PR waiting for review that fixes issues with this 
- https://github.com/sparkletown/sparkle/pull/2727 

(especially since required to ignore the anon mode among other things)

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:140]: Looks like out of place. Something like `{STRING_PLUS}` (can check `interpolationSettings.ts`) might make for clearer intent (if FA icon isn't that necessary)

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TableComponent/TableComponent.tsx:156]:
```suggestion
              WARNING: This action cannot be undone and will permanently remove
              {STRING_SPACE}
```

File: src/components/molecules/TablesUserList/TablesUserList.tsx
Comment (goran-peoski-work) [inline src/components/molecules/TablesUserList/TablesUserList.tsx:39]: shouldn't `template` be the more appropriate name since that's what we're using thorough the code base?

File: src/settings/index.ts
Comment (goran-peoski-work) [inline src/settings/index.ts:626]:
```suggestion

// These are really supposed to be constants and to avoid possible mutable shared state in the code elsewhere
Object.freeze(JAZZBAR_TABLES);
Object.freeze(CONVERSATION_TABLES);

```

File: src/utils/venue.ts
Comment (goran-peoski-work) [inline src/utils/venue.ts:39]: This can be improved later on with addition of a capabilities map, but for now it's good enough.

**Review by goran-peoski-work — COMMENTED** [review]
Looked good before, but since I found a typo, might as well have one last commit before merge

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: done 2ba4481

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: removed this thing completely 2ba4481

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: removed this reference as its unnecessary 2ba4481

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: redone using `&&` instead of ternary 2ba4481

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: replaced with `string_plus` constant 2ba4481

File: src/components/molecules/TableComponent/TableComponent.tsx
Comment (alexey-petrov) [reply src/components/molecules/TableComponent/TableComponent.tsx]: 2ba4481

File: src/components/molecules/TablesUserList/TablesUserList.tsx
Comment (alexey-petrov) [reply src/components/molecules/TablesUserList/TablesUserList.tsx]: yep, replaced 2ba4481

File: src/settings/index.ts
Comment (alexey-petrov) [reply src/settings/index.ts:626]: done 2ba4481

**Pull request merged**
